### DISCUSSION TITLE
How to create collections for pre-existing classes from external packages

### DISCUSSION THREAD

**Discussion by Sockitos** [isar/isar#502] [Help] [answered]
Hi,

I am currently using various packages for the different APIs my app requires.
I want to persist the data from those APIs but I don't know how to generate collections for the respective models, since they belong to external packages.
Is there a way to "extend" those models in order to use isar_generator to create the collections?

Thank you!

**Comment by simc** [comment, marked as answer]
As long as the property types of these models are compatible you can just extend them and the Isar generator should be able to pick up those properties.

**Reply by Sockitos** [reply]
Thanks! If they have nested objects and I want to "convert" them to links, is this doable while still extending the rest?

**Reply by simc** [reply]
That's going to be difficult. I'll release embedded object support soon which should make this much easier.

**Reply by Sockitos** [reply]
Ok, sounds awesome. I'm currently migrating some apps from hive to isar, so I guess I should wait for that update. Do you have an ETA for that version? Is it planned for this year?

**Reply by philos3** [reply]
https://github.com/isar/isar/blob/main/packages/isar_test/test/embedded_test.dart 
but i still can't use embedded
isar: 3.0.0-dev.8
isar_flutter_libs: 3.0.0-dev.8
